### DISCUSSION TITLE
Exit insert on timer

### DISCUSSION THREAD

**Discussion by favetelinguis** [meow-edit/meow#598] [Q&A] [answered]
Has anyone tried to setup so that you exit insert mode after some timeout. For example if I dont type a new char for 4 seconds I go back to the previous mode?

If I would like to try this setup what would the elisp code have to be for this to work nice?

The reasoning is that I much rather do a few exta i instead of ESC. And often I do small edit and think for a while before I do the next edit. I think this type of feature would save me from a ton ESC presses while having minimal impact on my typing. Not sure however how to set this up for testing the idea.

**Comment by eshrh** [comment, marked as answer]
I'm not sure if i'd really like this on my setup because I prefer to be intentional about stuff like this even if it costs me an extra keypress.

With that said, this isn't hard to implement:
```emacs-lisp
(add-hook
 'meow-insert-enter-hook
 (lambda ()
   (setq meow-insert-timer
         (run-with-idle-timer
          4 nil
          (lambda ()
            (when (eq meow--current-state 'insert)
              (meow--switch-state 'normal)))))))


(add-hook
 'meow-insert-exit-hook
 (lambda ()
   (when (and (bound-and-true-p meow-insert-timer)
              (timerp meow-insert-timer))
     (cancel-timer meow-insert-timer)
     (setq meow-insert-timer nil))))
```

**Reply by favetelinguis** [reply]
Wow thanks, amazing this will be fun to test!
